tests/unit/segments/test_segments_evaluator.py

```diff
@@ -148,12 +240,42 @@ def test_identity_in_segment_percentage_split(
     mock_get_hashed_percentage.return_value = identity_hashed_percentage
 
     # When
-    result = evaluate_identity_in_segment(identity=identity, segment=segment)
+    result = is_context_in_segment(context=context, segment=segment)
 
     # Then
     assert result == expected_result
 
 
+def test_context_in_segment_percentage_split__trait_value__calls_expected(
+    mocker: MockerFixture,
+    context: EvaluationContext,
+) -> None:
+    # Given
+    assert context["identity"] is not None
+    context["identity"]["traits"]["custom_trait"] = "custom_value"
+    percentage_split_condition = SegmentConditionModel(
+        operator=constants.PERCENTAGE_SPLIT,
+        value="10",
+        property_="custom_trait",
+    )
+    rule = SegmentRuleModel(
+        type=constants.ALL_RULE, conditions=[percentage_split_condition]
+    )
+    segment = SegmentModel(id=1, name="% split", rules=[rule])
```

Comment:
We should probably make sure that the structure of this segment looks like the ones created by flagsmith (a.k.a. the extra superfluous sandwich rule)